### DISCUSSION TITLE
Use existing Storage Account rather than creating new

### DISCUSSION THREAD

**Discussion by mmcdermo2** [actions/virtual-environments#4543] [Ideas]
I see that GenerateResourcesAndImage function is creating a new Storage Account each time it is invoked, and that is being passed as a parameter to packer.

Is there a reason why we can't add an optional parameter to this function so that users can have the VHD generated on their own specified Storage Account ?  Currently we're have to use AzCopy to copy the artifacts from the temporary Storage Account to a the existing Storage Account where we want them to reside.

It's not a huge deal, but would make our image generation process simpler.  However, I don't understand all the workings of how the entire repo is put together, so maybe there's a reason this can't be done.

If the idea above is practical however, I'd be happy to submit a PR

**Comment by al-cheb** [comment]
Hey, @mmcdermo2 
Feel free to submit a PR to update the GenerateResourcesAndImage function:

https://www.packer.io/docs/builders/azure/arm
**_To use an existing resource group you must provide:_**
- `build_resource_group_name (string)` - Specify an existing resource group to run the build in.

Providing `temp_resource_group_name` or `location` in combination with `build_resource_group_name` is not allowed.

E.g.:
```
packer build `
    -var "capture_name_prefix=$ResourcesNamePrefix" `
    -var "client_id=$ClientId" `
    -var "client_secret=$ClientSecret" `
    -var "install_password=$InstallPassword" `
    -var "resource_group=$ResourceGroup" `
    -var "build_resource_group_name=$BuildResourceGroup" `
    -var "subscription_id=$SubscriptionId" `
    -var "tenant_id=$TenantId" `
    -var "storage_account=$StorageAccount" `
    -var "run_validation_diskspace=$env:RUN_VALIDATION_FLAG" `
    $TemplatePath
```

**Comment by feliasson** [comment]
I also want the same type of functionality, currently I am cloning repo and loading GenerateResourcesAndImage function inside Azure Pipeline, thereafter I am doing string-replacements for the function before loading and running it to be able to specify my own storage account that will not be deleted. 

Please @al-cheb can you provide detailed variable explanation [here ](https://github.com/actions/virtual-environments/blob/main/docs/create-image-and-azure-resources.md#user-variables) for `build_resource_group_name` ?
Thank you
